packages/client/src/service/media/media.ts

```diff
@@ -0,0 +1,78 @@
+interface MP3Options {
+  volume: number;
+}
+const isMediaApiSupported = (mimeType: string, showError = true) => {
+  const isSupported =
+    'MediaSource' in window && MediaSource.isTypeSupported(mimeType);
+  if (showError && !isSupported) {
+    console.log('MediaSource is not supported');
+  }
+  return isSupported;
+};
+
+const audioElementMap = new Map<
+  string,
+  {
+    el: HTMLAudioElement;
+    mediaSource: MediaSource;
+  }
+>();
+
+function fetchAudio(url: string, cb: (resonse: any) => void) {
+  const xhr = new XMLHttpRequest();
+  xhr.open('get', url);
+  xhr.responseType = 'arraybuffer';
+  xhr.onload = function () {
+    cb(xhr.response);
+  };
+  xhr.send();
+}
+
+// Воспроизведение mp3 файла
+export const playMp3sound = (sourceUrl: string, options?: MP3Options) => {
+  if (isMediaApiSupported('audio/mpeg')) {
+    const mediaEl = audioElementMap.get(sourceUrl);
+    if (!mediaEl) {
+      audioElementMap.set(sourceUrl, {
+        el: document.createElement('audio'),
+        mediaSource: new MediaSource(),
+      });
+    }
+    if (!mediaEl) {
+      return;
+    }
```

Comment:
[Предложение]
Может объединить в один if?

```suggestion
    if (!mediaEl) {
      audioElementMap.set(sourceUrl, {
        el: document.createElement('audio'),
        mediaSource: new MediaSource(),
      });
      return;
    }
```

Reply:
Нет mediaEl всего будет определен, это чтобы TS не ругался

Reply:
Выходить из функции не надо, просто в настройках ts, стоит что нельзя указывать [имя переменной]!, поэтому такое условие